File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
        l, r := 0, len(numbers)-1
        var curr_sum int 
        for l <= r{
            curr_sum = numbers[l] + numbers[r]
            if  curr_sum == target{
                return []int{l+1, r+1}
            } else {
                if curr_sum > target{
                    r-=1
                } else{
                    l+=1
                }
            }
        }
        return []int{-1}
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers)-1
        while l <= r:
            curr_sum = numbers[l] + numbers[r]
            if  curr_sum == target:
                return [l+1, r+1]
            else:
                if curr_sum > target:
                    r-=1
                else:
                    l+=1
        return -1

